packages/util/src/obj.ts

```diff
@@ -15,135 +15,133 @@
  * limitations under the License.
  */
 
-// See http://www.devthought.com/2012/01/18/an-object-is-not-a-hash/
+type Keys<T> = Extract<keyof T, string>;
```

Comment:
I am huge fan of the changes in this file... but I feel that some of the methods here don't add very much value.  Have we considered getting rid of at least of some of them?

Reply:
I tried not to change anything that was working fine after my refactoring. If I remove these functions that can be replaced by native functions (and I want to), I'll just change all the packages in a single PR and get approval from everyone. So I'll leave these as is for now.

That's why I'm also leaving the util functions in your other comments instead of replacing them now. I'll send a PR right after this one.

Reply:
I merged the other PR into this one, so most of these functions are now gone.